lib/hiera/backend/vault_backend.rb

```diff
@@ -27,9 +27,16 @@ def initialize()
 
         begin
           @vault = Vault::Client.new
+          if @config[:prioritize_env]
```

Comment:
This isn't a very maintainable solution and it's only solving the problem for the specific config options *you* are trying to override. I think a more holistic and concise solution is necessary.